client/iotcore/client.go

```diff
@@ -206,6 +206,17 @@ func (c *client) UpsertDevice(ctx context.Context,
 				CertificateSigningRequest: aws.String(string(csr)),
 				SetAsActive:               *aws.Bool(device.Status == StatusEnabled),
 			})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	endpointType := "iot:Data-ATS"
```

Comment:
this should be a const, and probably moved to the top of the file

Reply:
If I make `endpointType` a const, I won't be able to pass it as pointer to `svc.DescribeEndpoint`

Reply:
IIRC, you can use aws.String, or something like that, which is designed to make pointers.

Reply:
Done!